src/cli/rustup_mode.rs

```diff
@@ -1379,14 +1378,46 @@ fn get_target(m: &ArgMatches, distributable: &DistributableToolchain<'_>) -> Opt
         .or_else(|| Some(distributable.desc().target.clone()))
 }
 
+fn get_component(
```

Comment:
I think it's fair. This function has been moved to `Component::try_new` by [5c6801d](https://github.com/rust-lang/rustup/pull/3601/commits/5c6801dca2d1e268afdf33168370550e15d61089).

Reply:
> I think it's fair. This function has been moved to `Component::try_new` by [5c6801d](https://github.com/rust-lang/rustup/pull/3601/commits/5c6801dca2d1e268afdf33168370550e15d61089).

@ongchi IIRC, since you added that function (?) and then moved it elsewhere in the same PR, maybe it's reasonable to squash this commit into the first one?

Reply:
With no further modifications, I just squash commits and update commit messages.